test/system/252-quadlet.bats

```diff
@@ -1139,4 +1139,90 @@ EOF
     run_podman image rm --ignore $image_for_test
 }
 
+@test "quadlet - image name" {
+    local quadlet_tmpdir=$PODMAN_TMPDIR/quadlets
+    local archive_file=$PODMAN_TMPDIR/archive-file.tar
+    local image_for_test=localhost/quadlet_image_test:$(random_string)
+
+    local quadlet_image_unit=image_test_$(random_string).image
+    local quadlet_image_file=$PODMAN_TMPDIR/$quadlet_image_unit
+    cat > $quadlet_image_file <<EOF
+[Image]
+Image=docker-archive:$archive_file
+ImageTag=$image_for_test
```

Comment:
Shouldn't this be another name than the one we used in `podman save`? 

I though that `ImageTag` would case the loaded/pull image to be tagged as such but I do not see the plumbing for it.  Am I missing something?

Reply:
Nice catch. I have the same question.

Reply:
Let me explain.

First, the lines:
```
run_podman image tag $IMAGE $image_for_test
run_podman image save --format docker-archive --output $archive_file $image_for_test
run_podman image rm $image_for_test
```
are the setup of the test, they are not the test itself.

After them, we are left with a docker-archive file saved under `$archive_file` that when pulled will translate to an image with the tag `$image_for_test` in the local container storage (`podman save` stores the tag as well). To make sure the archive file is in fact pulled, the setup removes the tag from the local container storage.

When the test runs, we have a dependency of container->volume->image, so starting the container will lead to the following:
Image Service: `podman image pull docker-archive:$archive_file`
Volume Service: `podman volume create --driver image -o image=$image_for_test`

The reason I needed the volume in this test is that `podman run` accepts `docker-archive:$archive_file` as the image to use and as a result, not setting `ImageTag` works as well. `podman volume create` however, does not support all transports. So, if `ImageTag` is not set, the command fails.

Reply:
Thanks for elaborating, @ygalblum.

So the problem is that the `.volume` is pointed to an `.image` where there latter does not expose an image reference that the `.volume` needs?

Reply:
Yes.
When a Quadlet file (`.container` or `.volume`) references a `.image` file, Quadlet translates the reference to the name of the resource (same as it does for `.volume` or `.network`). By default, the name of the resource produced by an `.image` file is the image passed in its `Image` key. However, in some cases, this does not work and for this reason, I added the new key `ImageTag`.

Reply:
I'd like to take a step back to fully understand the issue.

I think the source of the problem is that we conflated two things into one `Image` field: 1) an ordinary image reference, 2) an `.image` file.  Can we revisit 2)?  I think we are using 2) to extrapolate a systemd-depedency on the unit, are we?

Reply:
Quadlet does the same thing for `.network` and `.volume` files. For example, if in a `.container` file the user sets `Network=my-net.network`, Quadlet will notice that this is a reference to a `.network` Quadlet file and do two things:
1. Replace `my-net.network` with the name of the network. Originally, the network's name was determined by Quadlet to be `systemd-my-net`. We've since allowed the user to optionally set the Network Name (leaving the calculated name as the default behavior). Quadlet knows how to handle both cases.
2. Add a systemd dependency so that the `.service` created for the `.container` will depend on the `my-net-network.service`

The `.image` file provides the users with the exact same behavior. If a `.image` is referenced, Quadlet knows to make the appropriate changes. The current implementation handles the second task (adding dependency) without any issues. As for the first task, here we have a problem. 

In the current implementation, the name of the image, to which the `.image` reference is resolved, is the value of `Image` in the `.image` file. This works well when the value is an FQDN of an image in a registry. The service of the `.image` file pulls the image while service for both `.container` and `.volume` take the image from the local container storage.

The problem starts when the `Image` is of other type, e.g. `docker-archive`. In such case, the service of the `.image` file pulls the image into the local container storage. However, the behavior of the `.volume` and `.container` are not as expected:
1. `.container`: the call to `podman run` pulls the image from the archive file once again
2. `.volume`: the call to `podman volume create` fails since it does not support all container-transports and instead expects the image to be in the local container storage.

To address this issue, this change adds the `ImageTag` to allow the user to differentiate between what is being pulled and what will be used.

The example in the man page explain exactly that.

Reply:
>The example in the man page explain exactly that.

Almost, but not quite, and I think that was the source of my confusion. The phrasing is ambiguous: I interpreted "name of the image after being pulled" as "once the image is pulled, quadlet will rename/retag it with this new name". I now understand my mistake, but future readers might appreciate a clearer description. (I don't have a good suggestion right now, I'm sorry).

Reply:
Point taken. I thought the example explains it and I guess the root cause is that I'm explaining what I know without addressing what someone else does not. 
I'll welcome any rephrasing. Maybe @TomSweeneyRedHat, you usually make my doc stuff better

Reply:
Thanks, @ygalblum.  The additional `.container` example convinced me that's the right solution :+1:

Reply:
Maybe rename to `ExternalImageName`?

Reply:
Naming is hard -.-